core/src/test/java/io/grpc/internal/KeepAliveManagerTest.java

```diff
@@ -199,13 +204,8 @@ public void keepAlivePingFailsAfterItTimesOut() {
 
     // We do not receive the ping response. Shutdown runnable runs.
     ticker.time = 3000;
-    shutdown.run();
+    keepAlivePinger.onPingTimeout();
```

Comment:
Whoa. This needs to be `shutdown.run()`, otherwise it isn't really testing much.